rust/protocol/src/curve.rs

```diff
@@ -297,6 +297,16 @@ impl TryFrom<&[u8]> for PrivateKey {
     }
 }
 
+impl fmt::Debug for PrivateKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "PrivateKey {{ public_key_version={:?} }}",
```

Comment:
Nitpick: "version" sounds like a version number for me. Also, `public_key()` isn't *that* cheap, and I could see someone tossing `{:?}` into a hot log line without thinking about it. Maybe we should just log the key type instead.

Reply:
I think that's a fantastic idea and I'm making that change now!